tests/integration/aof-multi-part.tcl

```diff
@@ -0,0 +1,1065 @@
+            assert_equal 1 [is_alive $srv]
+            set client [redis [dict get $srv host] [dict get $srv port] 0 $::tls]
+            wait_done_loading $client
+
+            assert_equal v1 [$client get k1]
+            assert_equal v2 [$client get k2]
+            assert_equal v3 [$client get k3]
+        }
+    }
+
+    clean_aof_persistence $aof_dirpath
+
+    # Tests15: Multi Part AOF can load data when some AOFs are empty
+    create_aof_dir $aof_dirpath
+    create_aof $aof_base1_filepath {
+        append_to_aof [formatCommand set k1 v1]
+    }
+
+    create_aof $aof_incr1_filepath {
+    }
+
+    create_aof $aof_incr3_filepath {
+        append_to_aof [formatCommand set k3 v3]
+    }
+
+    create_aof_manifest $aof_manifest_filepath {
+        append_to_manifest "file appendonly.aof.1.base.aof seq 1 type b\n"
+        append_to_manifest "file appendonly.aof.1.incr.aof seq 1 type i\n"
+        append_to_manifest "file appendonly.aof.3.incr.aof seq 3 type i\n"
+    }
+
+    start_server_aof [list dir $server_path] {
+        test "Multi Part AOF can load data when some AOFs are empty" {
+            assert_equal 1 [is_alive $srv]
+            set client [redis [dict get $srv host] [dict get $srv port] 0 $::tls]
+            wait_done_loading $client
+
+            assert_equal v1 [$client get k1]
+            assert_equal "" [$client get k2]
+            assert_equal v3 [$client get k3]
+        }
+    }
+
+    clean_aof_persistence $aof_dirpath
+
+    # Tests16: Multi Part AOF can load data from old version redis
+    create_aof $aof_old_version_filepath {
+        append_to_aof [formatCommand set k1 v1]
+        append_to_aof [formatCommand set k2 v2]
+        append_to_aof [formatCommand set k3 v3]
+    }
+
+    start_server_aof [list dir $server_path] {
+        test "Multi Part AOF can load data from old version redis" {
+            assert_equal 1 [is_alive $srv]
+
+            set client [redis [dict get $srv host] [dict get $srv port] 0 $::tls]
+            wait_done_loading $client
+
+            assert_equal v1 [$client get k1]
+            assert_equal v2 [$client get k2]
+            assert_equal v3 [$client get k3]
+            
+            assert_equal 0 [check_file_exist $server_path $aof_basename]
+            assert_equal 1 [check_file_exist $aof_dirpath $aof_basename]
+           
+            assert_aof_manifest_content $aof_manifest_filepath  {
+                {file appendonly.aof seq 1 type b} 
+                {file appendonly.aof.1.incr.aof seq 1 type i}
+            }
+
+            assert_equal OK [$client set k4 v4]
+            
+            $client bgrewriteaof
+            while 1 {
+                if {[status $client aof_rewrite_in_progress] eq 1} {
+                    if {$::verbose} {
+                        puts -nonewline "\nWaiting for background AOF rewrite to finish... "
+                        flush stdout
+                    }
+                    after 1000
```

Comment:
that's a long sleep.
maybe this was copied from somewhere, so i'd like to use this opportunity to promote this into a utility function.
it can use shorter sleeps (10ms), and do the verbose puts only in one of 100 iterations

Reply:
Changed to waitForBgrewriteaof